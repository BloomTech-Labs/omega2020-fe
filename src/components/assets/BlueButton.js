import React from 'react';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { blue } from '@material-ui/core/colors';
import { withStyles } from '@material-ui/core/styles';

const BlueButton = (props) => {
  return (
    <ColorButton
      type={props.type}
      href={props.href}
      size='large'
      className='href'
      onClick={props.click}
    >
      <Typography variant='body2' className='title'>
        {props.title}
      </Typography>
    </ColorButton>
  );
};

const ColorButton = withStyles((theme) => ({
  root: {
    paddingLeft: 60,
    paddingRight: 60,
    paddingTop: 15,
    paddingBottom: 15,
    color: theme.palette.getContrastText(blue.A700),
    backgroundColor: blue.A700,
    '&:hover': {
      backgroundColor: blue[800],
    },
  },
}))(Button);

export default BlueButton;
