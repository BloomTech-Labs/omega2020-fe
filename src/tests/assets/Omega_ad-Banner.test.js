import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import OBanner from '../../components/assets/Omega_ad-Banner';
import BlueButton from '../../components/assets/BlueButton';
import Adapter from 'enzyme-adapter-react-16';
import { Typography, Box, CardHeader } from '@material-ui/core';
import AppIcon from '../../images/omega-mobile-icon.png';

configure({ adapter: new Adapter() });

describe('the Github banner ', () => {
  const ReactWrapper = mount(<OBanner />);

  it('snapshot the components', () => {
    const banner = renderer.create(<OBanner />).toJSON();
    const iBanner = banner.root;

    expect(banner).toMatchSnapshot();
    // expect(iBanner.find(BlueButton).props.title).toBe('GitHub')
    // expect(iBanner.findByProps({ className: 'title' }).children).toEqual(['Github']);
  });

  it('render the props', () => {
    const renderOBanner = renderer.create(<OBanner />); 
    const rootOBanner = renderOBanner.root;

    expect(rootOBanner.findByType(BlueButton).props.title).toBe('GitHub')
    expect(rootOBanner.findByType(BlueButton).props.href).toBe('https://github.com/Lambda-School-Labs/omega2020-fe')
})


  it('renders the material ui length correctly, and the text content', () => {
    expect(ReactWrapper.find(Typography)).toHaveLength(4);
    expect(ReactWrapper.find('h4').text()).toEqual('Omega 2020');
    expect(ReactWrapper.find('h5').text()).toEqual(
      'Coming soon to app stores near you :)'
    );
    expect(ReactWrapper.find(CardHeader)).toHaveLength(4);
    expect(ReactWrapper.find(Box)).toHaveLength(3);
  });

  it('render the image in the component', () => {
    expect(ReactWrapper.find('img')).toHaveLength(1);
    expect(ReactWrapper.find('img').prop('src')).toEqual(AppIcon);
  });
});
