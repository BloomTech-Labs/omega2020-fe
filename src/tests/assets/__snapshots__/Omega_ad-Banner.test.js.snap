// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Github banner  snapshot the components 1`] = `
<div
  className="MuiBox-root MuiBox-root-8 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-9 makeStyles-centerContainer-2"
  >
    <div
      className="MuiBox-root MuiBox-root-10 makeStyles-container-3"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <img
            alt="App Icon"
            className="img"
            src="omega-mobile-icon.png"
            style={
              Object {
                "width": "150px",
              }
            }
          />
        </div>
        <div
          className="MuiCardHeader-content"
        />
      </div>
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4"
          >
            Omega 2020
          </h4>
        </div>
      </div>
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Coming soon to app stores near you :)
          </h5>
        </div>
      </div>
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <a
              aria-disabled={false}
              className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-7 MuiButton-text href MuiButton-textSizeLarge MuiButton-sizeLarge"
              href="https://github.com/Lambda-School-Labs/omega2020-fe"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
            >
              <span
                className="MuiButton-label"
              >
                <p
                  className="MuiTypography-root title MuiTypography-body2"
                >
                  GitHub
                </p>
              </span>
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
